Vizzuality/gfw
# Feature/simplify analysis cached + fires whitelisting

## Overview

This updates the analysis cached service to be able to handle whitelist for `viirs` and `modis` for the new fires widgets. The `<WhitelistProvider/>` has been updated to now fetch all whitelist and the widget selectors to get the correct whitelist.

- An update to the analysis cached service to simplify the getting of the correct table ID. Now it is dynamically built from the fetch params `{dataset}_{location}_{datasetType}`
- Small changes to how the corresponding query build functions filter allowed params
- Reduced functions to build location query
- Small refactor to each getAnalysis function to reduce spreading of params and designate `dataset and` `datasetType` more explicitly.

## Testing

- Check fires widgets forest type and land category are now whitelisted
- Make sure all widget actually show data.
